### PR TITLE
refactoring executor for allowing for better validation WIP

### PR DESCRIPTION
**Pull request by chrislovecnm**
Currently, LifecycleExistsAndValidates and LifecycleExistsAndWarnIfChanges run until they timeout, which creates an odd UX experience and does not allow for allowing validation failures.

This PR fixes: https://github.com/kubernetes/kops/issues/3668

### CODE REVIEW

File: upup/pkg/fi/executor.go [hunk 1]
```diff
@@ -102,13 +103,31 @@ func (e *executor) RunTasks(taskMap map[string]Task, maxTaskDuration time.Durati
...
+				lifecycle = hl.GetLifecycle()
+			}
+
 			if err != nil {
 				remaining := time.Second * time.Duration(int(ts.deadline.Sub(time.Now()).Seconds()))
-				glog.Warningf("error running task %q (%v remaining to succeed): %v", ts.key, remaining, err)
 				errors = append(errors, err)
 				ts.lastError = err
+				if lifecycle == nil {
+					glog.Warningf("error running task %q (%v remaining to succeed): %v", ts.key, remaining, err)
```
Comment (andrewsykim) [inline upup/pkg/fi/executor.go]: This check is alredy done in `default` on the switch below

Comment (chrislovecnm) [reply upup/pkg/fi/executor.go]: Cool was not certain

Comment (andrewsykim) [reply upup/pkg/fi/executor.go]: hmm, actually you might get a nil pointer cause its *lifecycle :P

Comment (chrislovecnm) [reply upup/pkg/fi/executor.go]: Arg let me fix that above then

**Comment by chrislovecnm** [comment]
@justinsb need some help on this one, cause this did not work.  `kops update cluster us-west-2.aws.k8spro.com --phase cluster --yes` would not even try to create the vpc did not even die

**Comment by chrislovecnm** [comment]
@andrewsykim / @justinsb so I think I need a flag for dry-run and a flag for allowing fail on the executor.

I need to be able to test if I can skip the phase, but even still I need to know what type of task we have.  I will put this PR on the schedule for the office hours.

**Pull request closed without merging**